api/v1beta1/conversion.go

```diff
@@ -132,6 +132,42 @@ func Convert_v1beta1_ClusterClassSpec_To_v1beta2_ClusterClassSpec(in *ClusterCla
 	return nil
 }
 
+func Convert_v1beta1_MachineHealthCheckClass_To_v1beta2_MachineHealthCheckClass(in *MachineHealthCheckClass, out *clusterv1.MachineHealthCheckClass, s apimachineryconversion.Scope) error {
+	if err := autoConvert_v1beta1_MachineHealthCheckClass_To_v1beta2_MachineHealthCheckClass(in, out, s); err != nil {
+		return err
+	}
+
+	if len(in.UnhealthyConditions) > 0 {
```

Comment:
nit: can we drop the if?
(same across all conversions func)

Reply:
Done